lib/storage/tree.rb

```diff
@@ -0,0 +1,79 @@
+#==============================================================================
+# Copyright (C) 2022-present Alces Flight Ltd.
+#
+# This file is part of Flight Storage.
+#
+# This program and the accompanying materials are made available under
+# the terms of the Eclipse Public License 2.0 which is available at
+# <https://www.eclipse.org/legal/epl-2.0>, or alternative license
+# terms made available by Alces Flight Ltd - please direct inquiries
+# about licensing to licensing@alces-flight.com.
+#
+# Flight Storage is distributed in the hope that it will be useful, but
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, EITHER EXPRESS OR
+# IMPLIED INCLUDING, WITHOUT LIMITATION, ANY WARRANTIES OR CONDITIONS
+# OF TITLE, NON-INFRINGEMENT, MERCHANTABILITY OR FITNESS FOR A
+# PARTICULAR PURPOSE. See the Eclipse Public License 2.0 for more
+# details.
+#
+# You should have received a copy of the Eclipse Public License 2.0
+# along with Flight Storage. If not, see:
+#
+# For more information on Flight Storage, please visit:
+# https://github.com/openflighthpc/flight-storage
+#==============================================================================
+require "tty-tree"
+
+module Storage
+  class Tree
+    attr_reader :name, :children
+  
+    def initialize(name, children)
+      @name = name
+      @children = children
+    end
+    
+    # Convert to a hash format that tty-tree can render properly
+    def to_hash
+      hash_children = []
+      @children.each do |child|
+        if child.class == String
+          hash_children << child
+        else
+          hash_children << child.to_hash
+        end
+      end
+      return { @name => hash_children }
+    end
+    
+    def show
+      TTY::Tree[self.to_hash].render()

```

Comment:
I'm not sure how you have your Ruby environment set up, but I needed to add `tty-tree` to the Gemfile and `bundle install` before this would work.

Reply:
Oh right, I was testing with the gem installed just from the command line, fixed now.